src/audio/module_adapter/module_adapter.c

```diff
@@ -35,7 +35,21 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	int ret;
 	struct comp_dev *dev;
 	struct processing_module *mod;
-	struct ipc_config_process *ipc_module_adapter = spec;
+	unsigned char *data;
+	uint32_t size;
+
+	switch (config->type) {
+	case SOF_COMP_MODULE_ADAPTER:
+	{
+		struct ipc_config_process *ipc_module_adapter = spec;
+
+		size = ipc_module_adapter->size;
+		data = ipc_module_adapter->data;
+		break;
```

Comment:
@ranj063 there is a problem with this code. For codec_adapter we never used config->type. Does the codec_adapter tests still work for Intel usecases?

I'm confused about who sets config->type from the Linux kernel side.


Cc: @cujomalainey

Reply:
@dbaluta all components have their comp type set in the topology except the module_adapter comps. So we need to set that in topology and parse it in the kernel with ABI compatibility checks

Reply:
@ranj063 thanks for the explanation!